packages/angular/src/schematics/convert-tslint-to-eslint/convert.ts

```diff
@@ -0,0 +1,840 @@
+     *
+     * This is a worthwhile trade-off to avoid the need to force the user to manually run an install before
+     * invoking the schematic, and the dry-run output doesn't offer a ton of value for this use-case anyway.
+     */
+    if (
+      process.argv.includes('--dry-run') ||
+      process.argv.includes('--dryRun')
+    ) {
+      context.logger.error(
+        'NOTE: This schematic does not support --dry-run. If you are running this in Nx Console, it should execute fine once you hit the "Run" button.\n'
+      );
+      return process.exit(1);
+    }
+
+    if (!tree.exists('package.json')) {
+      throw new Error(
+        'Could not find a `package.json` file at the root of your workspace'
+      );
+    }
+
+    /**
+     * Ensure that the @nrwl/eslint-plugin-nx is installed
+     */
+    const packageJSON = tree.read('package.json').toString('utf-8');
+    const json = JSON.parse(packageJSON);
+
+    if (
+      !json.devDependencies?.['@nrwl/eslint-plugin-nx'] &&
+      !json.dependencies?.['@nrwl/eslint-plugin-nx']
+    ) {
+      const version: string | undefined =
+        json.devDependencies['@nrwl/workspace'];
+
+      context.logger.info(
+        '\nINFO: We are installing @nrwl/eslint-plugin-nx at the same version as your other @nrwl dependencies so that we can use it as part of the config generation...\n'
+      );
+
+      const pmc = getPackageManagerCommand();
+      execSync(`${pmc.addDev} @nrwl/eslint-plugin-nx@${version}`, {
+        stdio: 'ignore', // suppress any warnings about missing peerDeps to avoid confusion, they are all installed as part of the remaining schematic execution
+      });
+      context.logger.info(
+        '      Successfully installed @nrwl/eslint-plugin-nx\n'
+      );
+    }
+
+    const projectConfig = getProjectConfig(tree, schema.project);
+    const projectRoot = projectConfig.root;
+
+    // Expected existing files
+    const rootTSLintJsonPath = join(normalize(tree.root.path), 'tslint.json');
+    const projectTSLintJsonPath = join(normalize(projectRoot), 'tslint.json');
+
+    /**
+     * May or may not exist yet, depending on whether or not the schematic has been run
+     * before, or if this is already a mixed ESLint + TSLint workspace
+     */
+    const rootESLintrcJsonPath = join(
+      normalize(tree.root.path),
+      '.eslintrc.json'
+    );
+
+    return chain([
+      /**
+       * Handle adding ESLint and related dependencies and configuration
+       */
+      addAllConfigFilesAndCoreDependencies(projectRoot),
+      /**
+       * Ensure that the same TSLint config is not converted over and over.
+       */
+      removeExtendsFromProjectTSLintConfigBeforeConverting(
+        projectRoot,
+        projectTSLintJsonPath
+      ),
+      /**
+       * The conversion logic needs to be different for Angular projects vs e2e projects
+       * vs other project types which could use TSLint (such as NestJS).
+       */
+      isE2EProject(schema.project, projectConfig)
```

Comment:
It seems weird to me that we have logic within `@nrwl/angular` to handle other `@nrwl/*` things.

I think this generator belongs more in the `@nrwl/linter` package. I also think we may have a lot of utils in that package which will make it easier to work with lint configurations.

Yes, it will mean the user will have to install `@nrwl/linter` but maybe we can make that work to our advantage? :thinking:

I'm thinking we could make `convert-tslint-to-eslint` a dependency of `@nrwl/linter` which would mean we don't need an installation step in this generator.

What do you think?

Reply:
Why would we not need an installation step in the generator, sorry? It relies on eslint-plugin-nx, not linter, so I don't think anything about it works will change based on what you are suggesting?